File: api/PatoCup.Application/DTOs/Competition/MatchDto.cs
using System.ComponentModel.DataAnnotations;

namespace PatoCup.Application.DTOs.Competition
{
    public class CreateMatchDto
    {
        public int PhaseId { get; set; }
        public int Player1Id { get; set; }
        public int Player2Id { get; set; }
    }

    public class UpdateMatchDto
    {
        public int Id { get; set; }
        public int Player1Id { get; set; }
        public int Player2Id { get; set; }
        public int MatchStateId { get; set; }
        public int StateId { get; set; }
    }

    public class RegisterMatchResultDto
    {
        public int Id { get; set; }
        public int ScorePlayer1 { get; set; }
        public int ScorePlayer2 { get; set; }

        public int? WinnerId { get; set; } 
    }

    public class MatchResponseDto
    {
        public int Id { get; set; }

        public int PhaseId { get; set; }
        public string PhaseName { get; set; } = string.Empty;

        public int Player1Id { get; set; }
        public string Player1Name { get; set; } = string.Empty;

        public int Player2Id { get; set; }
        public string Player2Name { get; set; } = string.Empty;

        public int ScorePlayer1 { get; set; }
        public int ScorePlayer2 { get; set; }

        public int? WinnerId { get; set; }
        public string? WinnerName { get; set; } 

        public int MatchStateId { get; set; }
        public string MatchStateName { get; set; } = string.Empty;

        public int StateId { get; set; }
        public string StateName { get; set; } = string.Empty;
    }
}

File: client/src/app/data/features/tournaments/dtos/competition/match.dto.ts
export interface MatchResponseDto {
    id: number;
    phaseId: number;
    phaseName: string;
    player1Id: number;
    player1Name: string;
    player2Id: number;
    player2Name: string;
    scorePlayer1?: number;
    scorePlayer2?: number;
    winnerId?: string;
    winnerName?: string;
    matchStateId: number;
    matchStateName: string;
    stateId: number;
    stateName: string;
}

export interface CreateMatchDto {
    phaseId: number;
    player1Id: number;
    player2Id: number;
}

export interface UpdateMatchDto {
    id: number;
    player1Id: number;
    player2Id: number;
    matchStateId: number;
    stateId: number;
}

export interface RegisterResultDto {
    id: number;
    scorePlayer1: number;
    scorePlayer2: number;
    winnerId: number;
}